spec/services/preserved_object_handler_spec.rb

```diff
@@ -0,0 +1,384 @@
+          end
+          context 'error message' do
+            let(:result_msg) { results.select { |r| r[PreservedObjectHandler::DB_UPDATE_FAILED] }.first.values.first }
+
+            it 'prefix' do
+              expect(result_msg).to match(Regexp.escape(db_update_failed_prefix))
+            end
+            it 'specific exception raised' do
+              expect(result_msg).to match(Regexp.escape('ActiveRecord::ActiveRecordError'))
+            end
+            it "exception's message" do
+              expect(result_msg).to match(Regexp.escape('foo'))
+            end
+          end
+        end
+      end
+
+      it 'calls PreservedObject.save if the existing record is altered' do
+        po = instance_double(PreservedObject)
+        allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
+        allow(po).to receive(:current_version).and_return(1)
+        allow(po).to receive(:current_version=).with(incoming_version)
+        allow(po).to receive(:size=).with(incoming_size)
+        allow(po).to receive(:changed?).and_return(true)
+        allow(po).to receive(:save)
+        po_handler.update_or_create
+        expect(po).to have_received(:save)
+
+        allow(po).to receive(:destroy)
+      end
+      it 'calls PreservedObject.touch if the existing record is NOT altered' do
+        po_handler = described_class.new(druid, 1, 1)
+        po = instance_double(PreservedObject)
+        allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
+        allow(po).to receive(:current_version).and_return(1)
+        allow(po).to receive(:changed?).and_return(false)
+        allow(po).to receive(:touch)
+        po_handler.update_or_create
+        expect(po).to have_received(:touch)
+
+        allow(po).to receive(:destroy)
+      end
+      it 'logs a debug message' do
+        msg = "update #{druid} called and object exists"
+        allow(Rails.logger).to receive(:debug)
+        po_handler.update_or_create
+        expect(Rails.logger).to have_received(:debug).with(msg)
+      end
+    end
+
+    context 'druid not in db (yet)' do
+      after do
+        po = PreservedObject.find_by(druid: druid)
+        po.destroy if po
+      end
+      let(:po_handler) { described_class.new(druid, incoming_version, incoming_size) }
+      let(:exp_msg_prefix) { "PreservedObjectHandler(#{druid}, #{incoming_version}, #{incoming_size})" }
+      let(:exp_msg) { "#{exp_msg_prefix} added object to db as it did not exist" }
+
+      it 'creates the object' do
+        args = {
+          druid: druid,
+          current_version: incoming_version,
+          size: incoming_size,
+          preservation_policy: an_instance_of(PreservationPolicy)
+        }
+        allow(PreservedObject).to receive(:create).with(args)
+        po_handler.update_or_create
+        expect(PreservedObject).to have_received(:create).with(args)
+      end
+      it 'logs a warning' do
```

Comment:
Why should it warn that it created a new object? As opposed to Logger::INFO?

Reply:
That's what Lynn and Julian said at some story time or other.  See #90